### PR TITLE
Remove oauth refresh endpoint

### LINKED ISSUE #60: OAuthApplicationDatabase update_application infinite recursion
```
>>> db.update_application(**current)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ake/.venvs/ovos/lib/python3.11/site-packages/ovos_backend_client/database.py", line 425, in update_application
    self.update_application(oauth_service,
  File "/home/ake/.venvs/ovos/lib/python3.11/site-packages/ovos_backend_client/database.py", line 425, in update_application
    self.update_application(oauth_service,
  File "/home/ake/.venvs/ovos/lib/python3.11/site-packages/ovos_backend_client/database.py", line 425, in update_application
    self.update_application(oauth_service,
  [Previous line repeated 996 more times]
RecursionError: maximum recursion depth exceeded
```

The [update_application()](https://github.com/OpenVoiceOS/ovos-backend-client/blob/dev/ovos_backend_client/database.py#L421) method calls itself causing above exception.

What is the intention of this method? should it be forwarded to the standard update method or should it do additional checks?

### PR DESCRIPTION
**Pull request by forslund**
Remove refresh_endpoint as suggested in #59.

Since the change is breaking major version should have been updated but since it's pre 1.0.0 minor was updated in it's place

### CODE REVIEW

File: ovos_backend_client/version.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # The following lines are replaced during the release process.
 # START_VERSION_BLOCK
 VERSION_MAJOR = 0
-VERSION_MINOR = 1
+VERSION_MINOR = 2
 VERSION_BUILD = 1
 VERSION_ALPHA = 3
```
Comment (JarbasAl) [inline ovos_backend_client/version.py]: can you lower build to 0 and alpha to 1 , that will get automatically published on merge

**Comment by JarbasAl** [comment]
just realized most things in the wild are pinning 0.1.X , this change means those will be stuck with the recursion bug

can we break this into 2 PRs so those older versions can still get the bugfix, and then the removal of the endpoint marks the 0.2.0 breaking release ?

File: ovos_backend_client/version.py (hunk 1 above)
Comment (forslund) [reply ovos_backend_client/version.py]: So Minor 2, Build 0, alpha 1?

**Comment by forslund** [comment]
> just realized most things in the wild are pinning 0.1.X , this change means those will be stuck with the recursion bug
> 
> can we break this into 2 PRs so those older versions can still get the bugfix, and then the removal of the endpoint marks the 0.2.0 breaking release ?

Sure thing. I'll break them apart

**Comment by forslund** [comment]
Since there's been some movement around this I've taken the time to rebase so it's merging cleanly again.

**Comment by forslund** [comment]
Is this still of interest or should I close the PR?

**Comment by forslund** [comment]
^ ping @NeonDaniel @JarbasAl.

**Pull request merged**